### PR TITLE
Throw an error if a page is missing a description

### PR DESCRIPTION
**Pull request by parlough**
Throw an error if a page is missing a description with a new Liquid `throw_error` filter. This will prevent us adding new pages without descriptions in the future. As a result, I added descriptions to all pages currently missing one to get the build to pass.

I considered doing this with a [Jekyll hook](https://jekyllrb.com/docs/plugins/hooks/) instead, but that was finicky due to it considering all content types. This gives us easy control over which layouts want to make checks like this and makes it easy to add similar checks in the future.

Fixes https://github.com/flutter/website/issues/1176

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/development/ui/widgets/text.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ---
 title: Text widgets
 short-title: Text
+description: >
+  A catalog of Flutter's widgets for displaying and styling text.
```
Comment (sfshaza2) [inline src/development/ui/widgets/text.md:5]: I've never seen using the arrow for breaking lines in yaml text. But, in this case, it doesn't seem necessary.

File: src/security/index.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 ---
 title: Security
+description: >
+  An overview of the Flutter's team philosophy and processes for security.
```
Comment (sfshaza2) [inline src/security/index.md:4]: Same comment. Is the arrow needed here?

File: src/perf/power.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ---
 title: Preserving your battery
+description: Learn more about using limiting batter usage in your Flutter app.
+sitemap: false
```
Comment (sfshaza2) [inline src/perf/power.md:4]: "batter usage" ? LOL

But, maybe we should just delete these TBD pages?

File: src/development/platform-integration/ios/index.md [hunk 4]
```diff
@@ -1,4 +1,6 @@
 ---
 title: iOS
 layout: toc
+description: >
+  Content covering integration with iOS in Flutter apps.
```
Comment (sfshaza2) [inline src/development/platform-integration/ios/index.md:5]: Same comment as elsewhere.

File: src/perf/power.md (hunk 3 above)
Comment (parlough) [reply src/perf/power.md:4]: I've been wanting to get into baking!

Yeah they've been sitting in this state for a while. I'll delete them and make an issue tracking interest for those pages.

**Review by sfshaza2 — APPROVED** [review]
Wow, what an ambitious PR! You know, some years ago I searched for every page in the repo that had no description and added one, EXCEPT for the pages that were autogenerated, etc. :D. 

I didn't know that you could break lines up in the yaml matter with an arrow. I see you did that in many files, which I guess is fine. :) But I do think that the TBD pages should be deleted. And the "batter" page should move to the cookbook.  LOL

File: src/security/index.md (hunk 2 above)
Comment (parlough) [reply src/security/index.md:4]: In this case I used it to keep the line <= 80 characters.

**Pull request merged**